internal/provider/ovirt/ignition.go

```diff
@@ -77,43 +80,87 @@ func getOvirtClient(params *models.OvirtPlatform) (ovirtclient.Client, error) {
 		tls.CACertsFromMemory([]byte(*params.CaBundle))
 	}
 	if params.Fqdn != nil {
-		URL = fmt.Sprintf(engineURLStrFmt, *params.Fqdn)
+		URL = fmt.Sprintf(EngineURLStrFmt, *params.Fqdn)
 	}
 	if params.Username != nil {
 		userName = *params.Username
 	}
 	if params.Password != nil {
 		password = params.Password.String()
 	}
-	return ovirtclient.New(
+	ovirtClient, err := ovirtclient.New(
 		URL,
 		userName,
 		password,
 		tls,
 		nil,
 		nil,
 	)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create a new ovirt client")
+	}
+	p.OvirtClient = ovirtClient
+	return p.OvirtClient, nil
 }
 
-func updateHostInfoInManifest(clusterName string, vmName string, templateName string, workDir string, masterNum int) error {
-	vmNamePattern := fmt.Sprintf(vmNamePatternStrFmt, clusterName)
+func updateHostInfoInManifest(workDir, fileName, clusterName, vmName, templateName string) error {
+	vmNamePattern := fmt.Sprintf(VmNamePatternStrFmt, clusterName)
 	vmNameRegexp := regexp.MustCompile(vmNamePattern)
-	vmNameReplacement := fmt.Sprintf(vmNameReplacementStrFmt, vmName)
-	templateNameRegexp := regexp.MustCompile(templateNamePatternStr)
-	templateNameReplacement := fmt.Sprintf(templateNameReplacementStrFmt, templateName)
-
-	manifestFileName := fmt.Sprintf(manifestFileNameStrFmt, masterNum)
-	manifestPath := filepath.Join(workDir, "openshift", manifestFileName)
+	vmNameReplacement := fmt.Sprintf(VmNameReplacementStrFmt, vmName)
+	templateNameRegexp := regexp.MustCompile(TemplateNamePatternStr)
+	templateNameReplacement := fmt.Sprintf(TemplateNameReplacementStrFmt, templateName)
 
-	content, err := ioutil.ReadFile(manifestPath)
+	content, err := readContentFromFileGlob(workDir, fileName)
 	if err != nil {
-		return errors.Wrapf(err, "unable to read master file %s", manifestPath)
+		return errors.Wrapf(err, "unable to get contents from master machine file")
 	}
+
 	newContent := vmNameRegexp.ReplaceAllString(string(content), vmNameReplacement)
 	newContent = templateNameRegexp.ReplaceAllString(newContent, templateNameReplacement)
-	err = ioutil.WriteFile(manifestPath, []byte(newContent), 0600)
+
+	err = writeContentFromFileGlob(workDir, fileName, []byte(newContent))
+	if err != nil {
+		return errors.Wrapf(err, "unable to write contents to master machine file")
+	}
+	return nil
+}
+
+func readContentFromFileGlob(workDir, fileGlob string) ([]byte, error) {
+	globFullPath := filepath.Join(workDir, "openshift", fileGlob)
+	fullPaths, _ := filepath.Glob(globFullPath)
+
+	if fullPaths == nil {
+		return nil, errors.New(fmt.Sprintf("unable to find file '%s'", globFullPath))
+	}
```

Comment:
I think it's worth propagating the error, for cases where we have a malformed pattern